boards/xtensa/esp32/doc/esp32.rst

```diff
@@ -165,6 +168,30 @@ pinout.  Power to the ESP32 board should be provided by a USB cable.
 | IO15       | TDO       |
 +------------+-----------+
 
+Once the device is connected, you should be able to connect with (for
+a DevKitC board, replace with esp32-wrover.cfg for WROVER):
+
+.. code-block:: console
+
+    cd ~/esp/openocd-esp32
+    src/openocd -f interface/ftdi/flyswatter2.cfg -c 'set ESP32_ONLYCPU 1' -c 'set ESP32_RTOS none' -f board/esp-wroom-32.cfg -s tcl
+
+The ESP32_ONLYCPU setting is critical: without it OpenOCD will present
+only the "APP_CPU" via the gdbserver, and not the "PRO_CPU" on which
+Zephyr is running.  It's currently unexplored as to whether the CPU
+can be switched at runtime or if breakpoints can be set for
+either/both.
+    
+Now you can connect to openocd with gdb and point it to the OpenOCD
+gdbserver running (by default) on localhost port 3333.  Note that you
+must use the gdb distributed with the ESP-32 SDK.  Builds off of the
+FSF mainline get inexplicable protocol errors when connecting.
+    
+.. code-block:: console
+
+    ~/esp/xtensa-esp32-elf/bin/xtensa-esp32-elf-gdb outdir/esp32/zephyr.elf
+    (gdb) target remote localhost:3333
+
 Further documentation can be obtained from the SoC vendor in `JTAG debugging
 for ESP32`_.
```

Comment:
This link is broken because the **JTAG debugging for ESP32** was updated to V1.1 (2017.08.02) by Espressif [0], and now Espressif give us a on line documentation for this [1] instead a pdf file.

[0] http://espressif.com/en/support/download/documents?keys=&field_type_tid%5B%5D=13
[1] http://esp-idf.readthedocs.io/en/latest/api-guides/jtag-debugging/index.html